src/i_video.c

```diff
@@ -799,6 +799,9 @@ void I_FinishUpdate(void)
             {
                 I_Sleep((remaining_time - 1000) / 1000);
             }
+#if SDL_VERSION_ATLEAST(2, 24, 0)
+            SDL_CPUPauseInstruction(); // Play nicely with other threads
```

Comment:
Since this benefit applies to uncapped framerate, I'm curious if it would help with capped fps as well: https://github.com/fabiangreffrath/woof/blob/0f728d629b32457162637143f34dd70afc434a41/src/d_loop.c#L848

Before and after using `SDL_CPUPauseInstruction()`:
![Untitled-1](https://github.com/fabiangreffrath/woof/assets/56656010/6a5198ea-c639-41c3-88ed-07d7d28cedf1)

I'm not sure if it's correct to use it that way.

Reply:
Wow, your results are much better. Maybe it depends on the CPU, there is no difference on mine.

Reply:
> Maybe it depends on the CPU, there is no difference on mine.

I edited my comment with the exact change. It does seem to help with more consistent frame times, probably at the cost of more CPU use?

Reply:
> I edited my comment with the exact change

Oh, you are talking about the 35 FPS limit in d_loop.c, sorry I misunderstood. Yes that limiter is bad, but it's tied to the network code and that's a different issue.

Does this PR make a difference on your system?

Reply:
> Does this PR make a difference on your system?

No. There's no difference in frame times and no perceived difference in input latency.

Reply:
> No. There's no difference in frame times and no perceived difference in input latency.

What's the polling rate of your mouse?

Reply:
I'm really not sure what to think of this. Is `SDL_CPUPauseInstruction()` meant to be used *instead* of `I_Sleep()` or in addition to it?

Reply:
> What's the polling rate of your mouse?

It's a typical 1kHz mouse. What are you using? That reminds me, there's a related issue here for 8kHz mice: https://github.com/libsdl-org/SDL/issues/875 Despite the "2.30.0" milestone tag, I think it was only fixed for an upcoming "3.2.0" release, and the original linked issue goes back to 2021: https://github.com/libsdl-org/SDL/issues/4376 I wonder if this issue is present even with 1kHz mouse on older CPUs. Maybe this needs some profiling like this comment describes: https://github.com/libsdl-org/SDL/issues/4376#issuecomment-857319361

Reply:
> It's a typical 1kHz mouse. What are you using?

Same, 1kHz. I had seen that SDL issue as well. Suboptimal event handling + old CPU might be right. I can look into this some more when I get the chance (the joy_enable=0 issue possibly related). But I still think this `pause` change is good to have on its own.

Reply:
> I'm really not sure what to think of this. Is `SDL_CPUPauseInstruction()` meant to be used _instead_ of `I_Sleep()` or in addition to it?

So?

Reply:
> I'm really not sure what to think of this. Is `SDL_CPUPauseInstruction()` meant to be used _instead_ of `I_Sleep()` or in addition to it?

I think it should probably be used regardless of sleep (to avoid mis-speculating the loop exit). In the frame limiter code I see latency improvements adding the `pause` even when there are only two trips through the loop (like with @rfomin first suggested I_SleepUS() test). However, I didn't notice any latency issues with the 1ms sleeps like in d_loop.c, so for the sake of conservatism I didn't add the `pause` there.

Reply:
In fact I should probably move the `pause` to the top of the loop in order to avoid mis-speculating the early break where I_Sleep() doesn't get called at all.